### PR TITLE
Fix indexing for non-ascii text. Considerably accelerated processing …

### LINKED ISSUE #10: Unable to search through Chinese
Unable to search through Chinese, I hope you can help to fix it

### PR DESCRIPTION
**Pull request by anru**
…of html files

The main changes:
- Deleted cheerio: it incorrectly extracted non-ascii text
- Instead, [rehype](https://github.com/rehypejs/rehype) is used, which handles HTML processing perfectly.
- The most laborious work on processing HTML is carried out in a thread pool. It has accelerated processing of my documentation (lunr's index file takes 16 megabytes in my case) from 52 seconds to 15 seconds.
- lunr.client.js is now generated in the `@generated` directory for greater consistency. Writing files directly into a package in node_modules is not the most correct (and safe) approach.

This PR probably could_resolve #10 issue.
For me these changes fixes incorrect indexing russian language.

Also this PR fixes out of memory issue in case having a lot of documentation (my case):

```
docusaurus-lunr-search:: Building search docs and lunr index file
docusaurus-lunr-search:: writing search-doc.json

<--- Last few GCs --->

[41945:0x1045f3000]   197953 ms: Mark-sweep 2025.6 (2054.0) -> 2024.8 (2056.0) MB, 1253.9 / 0.0 ms  (average mu = 0.100, current mu = 0.003) allocation failure scavenge might not succeed
[41945:0x1045f3000]   199186 ms: Mark-sweep 2027.0 (2056.0) -> 2026.6 (2057.5) MB, 1227.9 / 0.0 ms  (average mu = 0.055, current mu = 0.004) allocation failure scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
```

### CODE REVIEW

File: src/html-to-doc.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    return
+  }
+  const markdown = select('.markdown', article)
+  if (!markdown) {
+    return
+  }
+
+  const pageTitleElement = select('h1', article)
+  if (!pageTitleElement) {
+    return
```
Comment (anru) [inline src/html-to-doc.js:42]: return here could lead to infinity loop. I'll fix it

**Comment by lelouch77** [comment]
@anru Thanks for you contribution 👍

**Comment by lelouch77** [comment]
@anru New version published [v2.1.1](https://github.com/lelouch77/docusaurus-lunr-search/releases/tag/v2.1.1)

**Comment by anru** [comment]
Thanks!

**Pull request merged**